domain/application/state/unit_test.go

```diff
@@ -955,3 +960,166 @@ func deptr[T any](v *T) T {
+
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		unitUUID.String(),
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.ARM64,
+		},
+	)
+	c.Check(err, jc.ErrorIs, applicationerrors.UnitNotFound)
+}
+
+// TestSetRunningAgentBinaryVersionNotSupportedArch tests that if we provide an
+// architecture that isn't supported by the database we get back an error
+// that satisfies [coreerrors.NotValid].
+func (s *unitStateSuite) TestSetRunningAgentBinaryVersionNotSupportedArch(c *gc.C) {
+	u := application.InsertUnitArg{
+		UnitName: "666",
+	}
+	s.createApplication(c, "666", life.Alive, u)
+	var unitUUID coreunit.UUID
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		return tx.QueryRowContext(ctx, "SELECT uuid FROM unit WHERE name = ?", u.UnitName).Scan(&unitUUID)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	unitUUID, err = s.state.GetUnitUUIDByName(context.Background(), coreunit.Name("666"))
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		unitUUID.String(),
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.Arch("noexist"),
+		},
+	)
+	c.Check(err, jc.ErrorIs, coreerrors.NotSupported)
+}
+
+// TestSetRunningAgentBinaryVersion asserts setting the initial agent binary
+// version (happy path).
+func (s *unitStateSuite) TestSetRunningAgentBinaryVersion(c *gc.C) {
+	u := application.InsertUnitArg{
+		UnitName: "666",
+	}
+	s.createApplication(c, "666", life.Alive, u)
+	var unitUUID coreunit.UUID
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		return tx.QueryRowContext(ctx, "SELECT uuid FROM unit WHERE name = ?", u.UnitName).Scan(&unitUUID)
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	unitUUID, err = s.state.GetUnitUUIDByName(context.Background(), coreunit.Name("666"))
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		unitUUID.String(),
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.ARM64,
+		},
+	)
+	c.Check(err, jc.ErrorIsNil)
+
+	var (
+		obtainedUnitUUID string
+		obtainedVersion  string
+		obtainedArch     string
+	)
+	err = s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		stmt := `
+SELECT unit_uuid,
+       version,
+       name
+FROM unit_agent_version
+INNER JOIN architecture ON unit_agent_version.architecture_id = architecture.id
+WHERE unit_uuid = ?
+	`
+
+		return tx.QueryRowContext(ctx, stmt, unitUUID).Scan(
+			&obtainedUnitUUID,
+			&obtainedVersion,
+			&obtainedArch,
+		)
+	})
+	c.Check(err, jc.ErrorIsNil)
+	c.Check(obtainedUnitUUID, gc.Equals, unitUUID.String())
+	c.Check(obtainedVersion, gc.Equals, jujuversion.Current.String())
+	c.Check(obtainedArch, gc.Equals, corearch.ARM64)
+}
+
+// TestSetRunningAgentBinaryVersion asserts setting the initial agent binary
```

Comment:
The function test name in the comment does not seem to match the test name

Reply:
Fixed